jetbrains-gateway/src/software/aws/toolkits/jetbrains/gateway/CawsConnectionProvider.kt

```diff
@@ -188,9 +189,18 @@ class CawsConnectionProvider : GatewayConnectionProvider {
                                     }
                                 }
                             } as ToolkitInstallSettings?
+                            val timeTakenToCheckInstallation = System.currentTimeMillis() - startTimeToCheckInstallation
                             toolkitInstallSettings ?: let {
                                 // environment is non-responsive to SSM; restart
                                 LOG.warn { "Restarting $envId since it appears unresponsive to SSM Run-Command" }
+                                CodecatalystTelemetry.devEnvironmentWorkflowStatistic(
```

Comment:
```suggestion

                            toolkitInstallSettings ?: let {
                                // environment is non-responsive to SSM; restart
                                val timeTakenToCheckInstallation = System.currentTimeMillis() - startTimeToCheckInstallation
                                LOG.warn { "Restarting $envId since it appears unresponsive to SSM Run-Command" }
                                CodecatalystTelemetry.devEnvironmentWorkflowStatistic(
```